### PR TITLE
PF-386: IntelliJ Inspections

### PR DESCRIPTION
**Pull request by jaycarlton**
Inspections for things like
* unused variables
* lambda style
* unused throws clauses (many more of these I need to come back and fix b/c autofix wasn't working)
* unboxing
* final
* stream efficiency
* static inner classes

Some other inspections are more nuanced, like Optionals used as parameters. That's controversial but largely harmless so I didn't rewrite those signatures.

### CODE REVIEW

File: src/main/java/bio/terra/workspace/app/controller/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -60,10 +61,10 @@
   }
 
   private ResponseEntity<ErrorReport> buildErrorReport(
-      Throwable ex, HttpStatus statusCode, List<String> causes) {
+      @NotNull Throwable ex, HttpStatus statusCode, List<String> causes) {
```
Comment (jaycarlton) [inline src/main/java/bio/terra/workspace/app/controller/GlobalExceptionHandler.java:64]: I added this b/c `ex.getMessage()` is flagged as possibly throwing an NPE. I can be talked out of it.

File: src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java [hunk 2]
```diff
@@ -203,8 +203,8 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
             workspaceId, DataReferenceType.fromApiModel(referenceType), name, userReq);
     logger.info(
         String.format(
-            "Got data reference %s in workspace %s for %s",
-            ref.toString(), referenceType, workspaceId.toString(), userReq.getEmail()));
+            "Got data reference %s of type %s in workspace %s for %s",
```
Comment (jaycarlton) [inline src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java:206]: wrong number of args for format

File: src/main/java/bio/terra/workspace/db/DataReferenceDao.java [hunk 3]
```diff
@@ -127,7 +128,7 @@ public boolean isControlled(UUID workspaceId, UUID referenceId) {
             .addValue("workspace_id", workspaceId.toString());
 
     try {
-      return jdbcTemplate.queryForObject(sql, params, Boolean.class).booleanValue();
+      return Optional.ofNullable(jdbcTemplate.queryForObject(sql, params, Boolean.class)).orElse(false);
```
Comment (jaycarlton) [inline src/main/java/bio/terra/workspace/db/DataReferenceDao.java:131]: possible NPE; unnecessary unboxing I think

File: workspace-manager-clienttests/src/main/java/scripts/testscripts/EnumerateDataReferences.java [hunk 4]
```diff
@@ -31,7 +29,6 @@
 
   private static final int DATA_REFERENCE_COUNT = 10;
   private static final int PAGE_SIZE = 4;
-  private List<UUID> dataReferenceIds;
```
Comment (jaycarlton) [inline workspace-manager-clienttests/src/main/java/scripts/testscripts/EnumerateDataReferences.java:34]: member can be local

**Review by ddietterich — APPROVED** [review]
OK, I scanned through these and they look reasonable.
Trust in the intelligence of the intellij!

**Review by zloery — APPROVED** [review]

File: src/main/java/bio/terra/workspace/service/crl/CrlService.java [hunk 5]
```diff
@@ -35,10 +35,9 @@
   public CrlService(CrlConfiguration crlConfig) {
     this.crlConfig = crlConfig;
 
-    final ClientConfig clientConfig;
     if (crlConfig.getUseCrl()) {
       GoogleCredentials creds = getApplicationCredentials();
-      clientConfig = buildClientConfig();
```
[the remark this answers is not in the document]
Comment (jaycarlton) [reply src/main/java/bio/terra/workspace/service/crl/CrlService.java]: 🐮

**Pull request merged**